### PR TITLE
Address bandit warnings: B602

### PR DESCRIPTION
**Pull request by athos-ribeiro**
To avoid the shell=True in the rpm query command, we query the RPM DB
with the RPM python bidings.  Note that the python 3 code path would
crash, since no stderr variable was being defined at all. This patch
consolidates the codepaths for both python 2 and 3, addressing the
related Bandit warning.

* https://bandit.readthedocs.io/en/latest/plugins/b605_start_process_with_a_shell.html
* OSBS-7453

Signed-off-by: Athos Ribeiro <athos@redhat.com>

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request includes link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

File: tests/plugins/test_koji_upload.py [hunk 1]
```diff
@@ -187,27 +184,41 @@ def taskFinished(self, task_id):
         return self.tag_task_state in ['CLOSED', 'FAILED', 'CANCELED', None]
 
 
-FAKE_SIGMD5 = b'0' * 32
-FAKE_RPM_OUTPUT = (
-    b'name1;1.0;1;' + B_LOCAL_ARCH + b';0;' + FAKE_SIGMD5 + b';(none);'
-    b'RSA/SHA256, Mon 29 Jun 2015 13:58:22 BST, Key ID abcdef01234567\n'
+class MockedRpmHeader(object):
```
Comment (MartinBasti) [inline tests/plugins/test_koji_upload.py]: This code seems duplicated

File: tests/plugins/test_koji_upload.py [hunk 2]
```diff
@@ -187,27 +184,41 @@ def taskFinished(self, task_id):
...
+                    v = '(none)'
+                return v
 
-    b'gpg-pubkey-doc;01234567;01234567;noarch;(none);' + FAKE_SIGMD5 +
-    b';(none);(none)\n'
 
-    b'name2;2.0;2;' + B_LOCAL_ARCH + b';0;' + FAKE_SIGMD5 + b';' +
-    b'RSA/SHA256, Mon 29 Jun 2015 13:58:22 BST, Key ID bcdef012345678;(none)\n'
-    b'\n')
+class MockedTS(object):
```
Comment (MartinBasti) [inline tests/plugins/test_koji_upload.py]: This seems duplicated too

File: tests/plugins/test_koji_upload.py (hunk 1 above)
Comment (athos-ribeiro) [reply tests/plugins/test_koji_upload.py]: Do you mean, duplication between koji_promote and koji_upload? I believe the duplication is intentional, while we still maintain the koji_promote plugin (which, IIRC, should be dropped in the future).

**Comment by athos-ribeiro** [comment]
- Re-based on master
- added python-rpm dependency on the spec file. Not adding to requirements.txt; see https://github.com/rpm-software-management/rpm/issues/273

File: tests/plugins/test_koji_upload.py (hunk 1 above)
Comment (MartinBasti) [reply tests/plugins/test_koji_upload.py]: Yes, I haven't realized that it is koji_promote. ack

**Comment by athos-ribeiro** [comment]
Rebased on master to get the  latest spec file changes in here and realized the test suite is broken due to a `flatpak_module_tools` update which requires a new version of `module-build-service`.

See https://pagure.io/flatpak-module-tools/c/52f878a91e4d5c42f25d8ed464461f32a538f538

**Comment by MartinBasti** [comment]
Waiving test suite failure, unrelated to this patch.

**Comment by MartinBasti** [comment]
Internal change only, no release notes needed

**Pull request merged**